ape_frame/accounts.py

```diff
@@ -34,25 +43,63 @@ def alias(self) -> str:
     def address(self) -> AddressType:
         return self.web3.eth.accounts[0]
 
-    def sign_message(self, msg: SignableMessage) -> Optional[MessageSignature]:
-        try:
-            raw_signature = self.web3.eth.sign(self.address, hexstr=_hash_eip191_message(msg).hex())
-        except ValueError as e:
-            if not e.args[0]["message"] == "User declined transaction":
-                raise
+    def sign_message(self, msg: Any, **signer_options: Any) -> Optional[MessageSignature]:
+        raw_signature = None
 
-            return None
+        if isinstance(msg, str):
+            # encode string messages as Ethereum Signed Messages
+            try:
+                raw_signature = self.web3.eth.sign(self.address, text=msg)
+            except ValueError as e:
```

Comment:
nit: PEP8 recommends avoiding single character variables names; i tend to agree `err` or `exc` is a better var name for these.